kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -98,6 +107,108 @@ public static Map<String, Map<String, Variant<?>>> buildSettings(NetworkProperti
+
+        Map<String, Variant<?>> settings = new HashMap<>();
+
+        // Configure Eap Method
+        switch (Kura8021xEAP.valueOf(eap)) {
+        case ttls:
+            build8021xTunneledTls(props, deviceId, settings);
+            break;
+        case peap:
+            build8021xProtectedEap(props, deviceId, settings);
+            break;
+        case tls:
+            build8021xTls(props, deviceId, settings);
+            break;
+        default:
+            throw new IllegalArgumentException("Security type 802-1x \"" + eap + "\" is not supported.");
+        }
+
+        if (!phase2.isPresent()) {
+            return settings;
+        }
+
+        // Configure Phase2 (innerAuth) Method
+        switch (Kura8021xInnerAuth.valueOf(phase2.get())) {
+        case none:
+            break;
+        case mschapv2:
+            build8021xMschapV2(props, deviceId, settings);
+            break;
+        default:
+            throw new IllegalArgumentException("Security type 802-1x \"" + phase2 + "\" is not supported.");
+        }
+
+        return settings;
+    }
+
+    private static void build8021xTunneledTls(NetworkProperties props, String deviceId,
+            Map<String, Variant<?>> settings) {
+        settings.put("eap", new Variant<>(new String[] { NM8021xEAP.ttls.name() }));
+        build8021xOptionalCaCertAndAnonIdentity(props, deviceId, settings);
+    }
+
+    private static void build8021xProtectedEap(NetworkProperties props, String deviceId,
+            Map<String, Variant<?>> settings) {
+        settings.put("eap", new Variant<>(new String[] { NM8021xEAP.peap.name() }));
+        build8021xOptionalCaCertAndAnonIdentity(props, deviceId, settings);
+    }
+
+    private static void build8021xTls(NetworkProperties props, String deviceId, Map<String, Variant<?>> settings) {
+        settings.put("eap", new Variant<>(new String[] { NM8021xEAP.tls.name() }));
+        build8021xOptionalCaCertAndAnonIdentity(props, deviceId, settings);
+
+        String identity = props.get(String.class, "net.interface.%s.config.802-1x.identity", deviceId);
+        settings.put("identity", new Variant<>(identity));
+
+        String clientCert = props.get(String.class, "net.interface.%s.config.802-1x.client-cert", deviceId);
+        settings.put("client-cert", new Variant<>(clientCert.getBytes(StandardCharsets.UTF_8)));
+
+        String privateKey = props.get(String.class, "net.interface.%s.config.802-1x.private-key", deviceId);
+        settings.put("private-key", new Variant<>(privateKey.getBytes(StandardCharsets.UTF_8)));
+
+        String privateKeyPassword = props
+                .get(Password.class, "net.interface.%s.config.802-1x.private-key-password", deviceId).toString();
+        settings.put("private-key-password", new Variant<>(privateKeyPassword));
+
+    }
+
+    private static void build8021xOptionalCaCertAndAnonIdentity(NetworkProperties props, String deviceId,
```

Comment:
Rename this method for consistency. We reserved `build*` for the top level public methods, `create*` for the private ones called by the `build*` ones.

```suggestion
    private static void create8021xOptionalCaCertAndAnonIdentity(NetworkProperties props, String deviceId,
```